ansible_base/oauth2_provider/serializers/application.py

```diff
@@ -0,0 +1,92 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils.translation import gettext_lazy as _
+from oauth2_provider.generators import generate_client_secret
+
+from ansible_base.lib.serializers.common import NamedCommonModelSerializer
+from ansible_base.lib.utils.encryption import ENCRYPTED_STRING
+from ansible_base.oauth2_provider.models import OAuth2Application
+
+
+class OAuth2ApplicationSerializer(NamedCommonModelSerializer):
+    oauth2_client_secret = None
+
+    class Meta:
+        model = OAuth2Application
+        fields = NamedCommonModelSerializer.Meta.fields + [x.name for x in OAuth2Application._meta.concrete_fields]
+        read_only_fields = ('client_id', 'client_secret')
+        read_only_on_update_fields = ('user', 'authorization_grant_type')
+        extra_kwargs = {
+            'user': {'allow_null': True, 'required': False},
+            'organization': {'allow_null': False},
+            'authorization_grant_type': {'allow_null': False, 'label': _('Authorization Grant Type')},
+            'client_secret': {'label': _('Client Secret')},
+            'client_type': {'label': _('Client Type')},
+            'redirect_uris': {'label': _('Redirect URIs')},
+            'skip_authorization': {'label': _('Skip Authorization')},
+        }
+
+    def _get_client_secret(self, obj):
+        request = self.context.get('request', None)
+        try:
+            if obj.client_type == 'public':
+                return None
+            elif request.method == 'POST':
+                if self.oauth2_client_secret is None:
+                    # This should be an impossible case, but...
+                    return ENCRYPTED_STRING
```

Comment:
Is this right? Would we want to return `ENCRYPTED_STRING` if we didn't have a secret? Seems like it could mask an issue.

Reply:
I removed this case, so now it'll just show the `null` if it's `None`.